deepsensor/data/loader.py

```diff
@@ -797,6 +829,66 @@ def sample_offgrid_aux(
             Y_t_aux = Y_t_aux.reshape(1, *Y_t_aux.shape)
         return Y_t_aux
 
+    def sample_patch_size_extent(self) -> Sequence[float]:
+        """Sample patch size.
+
+        :return sequence of patch spatial extent as [lat_min, lat_max, lon_min, lon_max]
+        """
+        # assumption of normalized spatial coordinates between 0 and 1
+
+        lat_extend, lon_extend = self.patch_size
+
+        lat_side = lat_extend / 2
+        lon_side = lon_extend / 2
+
+        # sample a point that satisfies the boundary and target conditions
+        continue_looking = True
```

Comment:
I would remove the `continue_looking` logic entirely. Firstly, it's fine if the patch contains no context data; DeepSensor models should be able to handle this. The main risk here is that the patch contains no target data, which can lead to NaNs when passed to the `ConvNP.loss_fn`. However, it is much, much, easier to check for `Task`s with no target data as a training pre-processing step. This would be a separate PR or something we expect the user to be aware of.

Reply:
Is there an assumption about a common coordinate range between the context and the target? Because if so, we can gather the coordinate bound extend of the target variable and use that to do the random window sampling?

Reply:
No, unfortunately we can't assume that. We'll have to loop over all the `self.context` and `self.target` variables updating the min/max data coordinate bounds.